### DISCUSSION TITLE
Thoughts on suggestions display

### DISCUSSION THREAD

**Discussion by romzie** [marlonrichert/zsh-autocomplete#207] [Ideas]
### Thought 1

**Current behavior :**
When too much suggestions, nothing is displayed. Pressing tab enables a partial list of suggestions.

**Proposition :**
Add a message to notify user that there is actually some suggestions but too much to fit in screen.
Something like "Too much (X) items to display."

**My thoughts :**
This plugin always almost displays something so it felt weird having no information on current directory content for the use case `cd /usr/share/` which had a lot of stuff inside.

### Thought 2

**Current behavior:**
Height size of suggestions list seems to be pre-determined (I suppose as a function of the terminal height) and is quite big.

**Proposition :**
Enable custom height of suggestions list via zstyle.
Might be a percent of terminal height, or absolute but it might be easier to implement percent.

**My thoughts :**
As of personnal preference, It would be great to set height of list to like 50% of terminal height so I can see more of my command history when the suggestion list is long.

### Thought 3

**Current behavior :**
Same as thought 1 : When too much suggestions, nothing is displayed. Pressing tab enables a partial list of suggestions.

**Proposition :**
(Could be an option to enable)
Display a partial list of suggested items with a bottom line saying that there is more which is not displayed.

**My thoughts :**
Well I went into this plugin to have suggestions at all time and sometimes my terminal is too small or there is too much items to display, but I wish I add suggestions anyway.

_Let's hear other users thoughts on this and tell me if something is unclear !_

**Comment by marlonrichert** [comment]
1. I'm already working on this. Stay tuned. 🙂 
2. Good idea. I will implement it.
3. Not doable, unfortunately. At least not without a significant performance hit. But I could enable it as an option, for those who don't mind a bit slower completion when this happens. 🤔

**Comment by romzie** [comment]
1. & 2. Wow nice, thank you !
3. I am curious about this. When pressing tab when the list is too long, zsh provides me with the full list, but a partial view over it (you can go in the list and actually see there is more than the terminal can display). I know nothing about zsh but what are some technical issues to reproduce this ?

**Comment by marlonrichert** [comment]
> When pressing tab when the list is too long, zsh provides me with the full list, but a partial view over it (you can go in the list and actually see there is more than the terminal can display). I know nothing about zsh but what are some technical issues to reproduce this ?

I'm not familiar with this behavior. How can I reproduce it? Can you share screenshots of what it looks like?

**Comment by romzie** [comment]
Made a little gif [here](https://gfycat.com/whimsicalhatefulamericancurl)
so plugin is enabled (note that i dont use fzf), suggestions are displayed fine until I reach /usr/share/ which contains too much items to display, I then press tab to enable a list that is displayed even though it is longer than the terminal window, which I guess is created by zsh

**Comment by marlonrichert** [comment]
Ah, you mean menu selection. I've tried that, but it doesn't work in this case, because it automatically inserts the first item from the menu into the command line, thus disrupting what you've typed so far. 🙂

**Comment by marlonrichert** [comment]
I pushed in an setting to set the completion list on the [`dev` branch](https://github.com/marlonrichert/zsh-autocomplete/tree/dev) (which also includes asynchronous completion). Please checkout that branch to test the feature and let me know what you think. 🙂

**Comment by romzie** [comment]
Really really great what youy did with the possibility to enter the menu selection, love it !
Also tested the max height setting which works fine too. Very good job !
